### PR TITLE
Allow Cloudformation use file contents directly

### PR DESCRIPTION
**Pull request by benbarclay**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

cloudformation
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file =
  configured module search path = Default w/o overrides
```
##### SUMMARY

This allows a user to give the cloudformation module a string that contains the contents of a cloudformation template.
This allows a user to use a lookup plugin to ensure that a cloudformation role can be distributed as a plugin without having to hardcode the full path, or upload to a remote s3 bucket.

This was discussed in [1460](https://github.com/ansible/ansible-modules-core/issues/1460)

To be entirely honest, not really sure how to test this....

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @benbarclay. To the current maintainers, @jsmartin please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/cloudformation.py [hunk 1]
```diff
@@ -143,6 +148,21 @@
       ClusterSize: 3
     tags:
       Stack: ansible-cloudformation
+
+# Use a template with a lookup
+- name: launch ansible cloudformation example
+  cloudformation:
+    stack_name="ansible-cloudformation" state=present
+    region=us-east-1 disable_rollback=true
+    template_contents="{{ lookup('file', 'cloudformation.json') }}"
```
Comment (ryansb) [inline cloud/amazon/cloudformation.py:157]: This example would actually fail because the JSON file contents would be parsed into a Python type.

**Comment by ryansb** [comment]
I'm not sure adding this option is the way to go. inline JSON is usually a pain (see https://github.com/ansible/ansible/issues/7005#issuecomment-42894813 ) and the `iam_policy` module's `policy_json` option. 

Maybe a better way would be to use a variable to store the template path, so the existing option could still be used. You could even use the `template` lookup to make conditional decisions about the file path.

**Comment by benbarclay** [comment]
Makes sense to me. I'll close this ticket.

**Pull request closed without merging**